enhancements/oc/cli-manager.md

```diff
@@ -0,0 +1,252 @@
+from the location/image in each custom resource.
+
+CLIs could be installed in a similar manner to plugins through [krew](https://github.com/kubernetes-sigs/krew).
+CLI images would serve the artifacts to enable disconnected downloads, and OpenShift CLIs will be available as
+[krew plugins](https://github.com/kubernetes-sigs/krew-index/tree/master/plugins).
+
+* Krew and Krew plugins are upstream projects that Kubernetes users are already familiar with
+* The index will be served by the CRD controller with its contents managed by cluster-admins via CRs
+* A Krew-compatible index can provide available CLIs and plugins for a cluster (similar to the current index) and can also provide an easy mechanism for installing through `oc tools`
+* An openshift-krew-index with supported artifacts could hold information about CLIs supported by OpenShift.
+* With `krew index add https://someother-third-party-index` we won't limit cluster-admins from adding their own index with whatever plugins they want. Only the openshift-krew-index will be supported.
+
+Users will continue to have the option to download binaries from either the central location where artifacts are published or from the console.
+This proposal adds the mechanism to install from images built from the same artifacts using `oc`. All options will offer the same binaries,
+ensured through sha256 signatures.
+
+For the initial implementation, supported CLIs will create ClusterCLI custom resources.  The CLI will be installed to a user's home directory with
+the same logic as `oc image extract` and from each CLI's artifacts image (see oc, cli-artifacts image for reference).
+Through images and ImageContentSourcePolicy, CLIs will be available to disconnected environments. Currently, the console provides
+oc binaries by running a webserver pod in the console namespace with a route.  The cli-manager will take over running this webserver, this will
+be included in the CRD controller that will create the core ClusterCLI custom resources (oc, odo, and helm, the 3 CLIs currently available
+through the consoleclidownloads mechansism).
+
+### User Stories
+
+#### Story 1
+
+As a user, I want a CLI manager for various CLIs available for OpenShift and related services so that I can discover, install and list them
+The user will invoke the following commands:
+
+* `oc tools list` will access the connected OpenShift cluster, if connected,  and will retrieve information about available CLIs (ClusterCLI CRs)
+* `oc tools install odo` will install `odo` to a user's home directory, as a standalone binary.
+
+##### Example
+```text
+$ oc tools list
+NAME                 DESC                        LATEST       INSTALLED
+-----                -----                       -----        -----
+kubectl              Kubernetes CLI              1.15         1.13
+oc                   OpenShift CLI               4.4          4.3
```

Comment:
How will `oc` know which version of the tool is installed locally? Will it ask each of them to report their version? Where does the info from the latest column come from?

Reply:
Great questions...keeping track of locally installed version was an interesting problem. Since a SHA256 hash is kept on every version/os/arch binary, `oc` can simply hash the binary locally and query the API with that digest to get the current version. The digests are stored in the CRs status field, so as long as no one removes those, `oc` should always know what version is installed.

In the CR, it is expected that the list of versions provided is in order from oldest to newest, so the latest is always at the end of the list. Attempting to parse version numbers seemed too problematic given how different projects handle versioning.

Reply:
Either explain the questions popping up in proposal section, or add a questions section towards the end of the proposal, It's fine and good to even leave some questions unaswered

Reply:
Also, doesn't krew track that information for you?